### PR TITLE
13 utilization reports

### PR DESCRIPTION
**Pull request by kmagida110**
## Background
This PR does the work to combine Site data and student data through Facility Codes to make Utilization reports. The previous join was imperfect and left out some rows. Utilization reports are important to OEC because it shows them sites that are allocated spaces and where there might be opportunities for additional slots to be filled. 

Unfortunately, after actually joining and validating the data there are some large discrepancies in the data for many sites between the number of allocated sites and the enrolled students. The chart that would be used is here: https://oecdata.org/superset/explore/?form_data=%7B%22slice_id%22%3A%2076%7D

- Data quality
  - The allocated space data is likely accurate but is definitely stale now and OEC does not have a centralized place to store these
  - Some of the programs that are the mo
- It seems like some providers input all students who attended the program in a year, if there was turnover then multiple students would have been taking the same slot.  

To state the obvious, this is why we built ECE Reporter and as we get more data from them we'll be able to get more accurate reports. Getting allocated spaces from ECE Reporter and having time series data will make this a much more effective report. As is with the July 2020 data, it is a reasonable look as a POC if we filter out sites with apparent > 100% utilization. In reality, my recommendation is we should do this ticket again with ECE Reporter data when we have more control over the inputs.

## GitHub Issue
https://github.com/ctoec/data-visualization/issues/13

## Validation Plan
Validate my assumption that the chart above is not something we want in the main dashboard right now and that assumptions above are correct.

## Additional Context
This builds off the work here: https://github.com/ctoec/data-visualization/pull/20 so that PR will need to be approved first.

### CODE REVIEW

**Comment by JordanGuinn** [comment]
Can you either squash-merge https://github.com/ctoec/data-visualization/pull/20 (once validated of course) to `main` and merge that in here, or repoint this PR to the `legislative-districts` branch?  Just so we can see what the actual diff here is

**Comment by kmagida110** [comment]
@JordanGuinn this is ready to review now, I merged in  https://github.com/ctoec/data-visualization/pull/20

File: src/data_integration/july_2020/build_tables.py [hunk 1]
```diff
@@ -16,10 +16,20 @@ def standardize_facility_string(col: pd.Series) -> pd.Series:
     :param col: dataframe column with facility codes
     :return: column of renamed facility codes
     """
+    def clean_string(item):
+        item = str(item).split(' or ')[0]
```
Comment (jlhogan) [inline src/data_integration/july_2020/build_tables.py:20]: explain what this does / why necessary ?

File: src/data_integration/july_2020/build_tables.py [hunk 2]
```diff
@@ -16,10 +16,20 @@ def standardize_facility_string(col: pd.Series) -> pd.Series:
...
+        item = item.replace('Annex to', '').replace('?', '').strip()
 
-    # Add FC to Facility Code column and pad to 10 digits to ensure DB reads it correctly
-    return 'FC' + col.str.zfill(10)
+        # Some facility codes were entered in correctly, this fixes the errors
+        if item in SITE_FACILITY_LOOKUP_DICT:
+            item = SITE_FACILITY_LOOKUP_DICT[item]
+        # Add FC to Facility Code column and pad to 10 digits to ensure DB reads it correctly
```
Comment (jlhogan) [inline src/data_integration/july_2020/build_tables.py:26]: if we're deciding that facility codes should be prefixed with FC, should we make that change to the data in ECE Reporter as well ?

File: src/data_integration/july_2020/build_tables.py [hunk 3]
```diff
@@ -203,16 +213,43 @@ def merge_legislative_data(student_df):
...
+    :param join_col: Column to join legislative district to dataframe
+    :return: pd.Dataframe with additional data on state legislators
+    """
+    with open(SITE_LEGIS_LOOKUP, 'r') as f:
+        leg_lookup = json.load(f)
+
+    rows = []
+    for _, row_dict in leg_lookup.items():
+
+        # Get metadata from legislative lookup that is not from the API call
```
Comment (jlhogan) [inline src/data_integration/july_2020/build_tables.py:229]: what does this mean?

File: src/data_integration/july_2020/build_tables.py [hunk 4]
```diff
@@ -203,16 +213,43 @@ def merge_legislative_data(student_df):
...
+        initial_data = {key: row_dict[key] for key in row_dict if key != 'raw_result'}
+
+        # Get the API call with the specified columns
```
Comment (jlhogan) [inline src/data_integration/july_2020/build_tables.py:232]: what does this mean?

File: src/data_integration/july_2020/constants.py [hunk 5]
```diff
@@ -66,6 +65,10 @@
     'ZIP [ECE]': 'ZIP Code'
 }
 
+SITE_FACILITY_LOOKUP_DICT = {'0280411 / 0280341': '280341',
```
Comment (jlhogan) [inline src/data_integration/july_2020/constants.py:68]: can we note where this comes from // what it's for?

File: src/data_integration/july_2020/build_tables.py (hunk 1 above)
Comment (kmagida110) [reply src/data_integration/july_2020/build_tables.py:20]: Done

File: src/data_integration/july_2020/build_tables.py (hunk 2 above)
Comment (kmagida110) [reply src/data_integration/july_2020/build_tables.py:26]: No, I'm hoping that the process for pulling data from ECE into the data viz tool will either a. not involve CSV uploads and therefore be more precise or b. if it does involve CSV uploads and the matching issue happens again we can add it in a data transformation.

File: src/data_integration/july_2020/build_tables.py (hunk 3 above)
Comment (kmagida110) [reply src/data_integration/july_2020/build_tables.py:229]: I added clarification but it pulls the lat, long and Facility Code that were associated with the raw results from the API stored in the lookup

File: src/data_integration/july_2020/build_tables.py (hunk 4 above)
Comment (kmagida110) [reply src/data_integration/july_2020/build_tables.py:232]: It pulls the stored json from the geo matching API and parses out the fields that we want, updated the docs

File: src/data_integration/july_2020/constants.py (hunk 5 above)
Comment (kmagida110) [reply src/data_integration/july_2020/constants.py:69]: Yup, added that, it was a manual process.

**Comment by kmagida110** [comment]
@jlhogan ready for a re-review

**Review by JordanGuinn — APPROVED** [review]

**Review by bamader — APPROVED** [review]

**Pull request merged**